src/coreComponents/functions/TableFunction.cpp

```diff
@@ -162,6 +162,23 @@ void TableFunction::reInitializeFunction()
   m_kernelWrapper = createKernelWrapper();
 }
 
+void TableFunction::checkCoord( real64 const coord,
+                                localIndex const dim,
+                                char const * const dimName,
+                                char const * const tableName ) const
+{
+  GEOS_THROW_IF( dim >= m_coordinates.size(),
```

Comment:
can `dim` be < 0 ?

Reply:
One should never request to check a coord in a negative dimension index, but I added a check as the `localIndex` type allows negative values:
```suggestion
  GEOS_THROW_IF( dim >= m_coordinates.size() || dim < 0,
```

I used localIndex as this is the default type used for sizes for arrays. I don't know why `size_t` or `uint` are not prefered.